python/dgl/graphbolt/itemset.py

```diff
@@ -255,6 +267,44 @@ def __iter__(self) -> Iterator:
     def __len__(self) -> int:
         return sum(len(itemset) for itemset in self._itemsets.values())
 
+    def __getitem__(self, idx: Union[int, slice, Iterable]) -> Dict[str, Tuple]:
+        try:
+            len(self)
+        except TypeError:
+            raise TypeError(
+                f"{type(self).__name__} instance doesn't support indexing."
+            )
+        offsets = [len(itemset) for itemset in self._itemsets.values()]
+        total_num = sum(offsets)
+        if isinstance(idx, int):
+            if idx < 0:
+                idx += total_num
+            assert idx < total_num, (
+                f"Index {idx} out of range for {type(self).__name__} "
+                f"instance with length {total_num}."
+            )
+            for key, offset in zip(self._itemsets.keys(), offsets):
```

Comment:
make offset prefix sum and use binary search?

Reply:
the main reason I choose this implementation is to be easy to handle range that across multiple keys. and the complexity is trivial and binary search does not bring obvious improvement in most cases.

Reply:
oh, I was wrong. in this case, yes. binary search is better

Reply:
BS is applied

Reply:
But I'm not sure it really perform better than original one.